Automattic/wp-calypso
# Payment Methods: add PayPal to change payment method form

#### Changes proposed in this Pull Request

This adds a PayPal button to the list of options displayed by `ChangePaymentMethod`. Selecting this option will create a new PayPal billing agreement for the subscription in question.

![Screen Shot 2021-01-07 at 3 12 05 PM](https://user-images.githubusercontent.com/9310939/103945492-e3deaf00-50fa-11eb-9ee2-6b882ca876e0.png)

Fixes https://github.com/Automattic/wp-calypso/issues/47518

#### Testing instructions

- Apply D54805-code 
- Visit `/purchases` for a site which has at least one subscription.
- Click on the subscription.
- Click "Add payment method" or "Change payment method"
- Verify that "PayPal" is one of the radio button options. Click to select PayPal.
- Click the submit button at the bottom of the form.
- Verify that you are redirected to PayPal to create a new billing agreement.
- Refuse the billing agreement and verify that you are redirected back to the form to select a payment method.
- Click to select PayPal again.
- Confirm the PayPal billing agreement this time and verify that you are redirected back to the subscription page.
- Verify that you see a notification that says your payment method has been assigned.
- Verify that the payment method assigned to the subscription is PayPal (you should see it in the subscription's card).
